packages/expo-battery/src/ExpoBattery.web.ts

```diff
@@ -70,18 +70,37 @@ export default {
   },
 };
 
-function onChargingChange(this: BatteryManager): void {
-  const batteryState = this.charging ? BatteryState.CHARGING : BatteryState.UNPLUGGED;
+let lastReportedState: BatteryState = BatteryState.UNKNOWN;
+
+function getBatteryState(isCharging: boolean, level: number) {
+  return isCharging
+    ? level >= 1.0
+      ? BatteryState.FULL
+      : BatteryState.CHARGING
+    : BatteryState.UNPLUGGED;
+}
+
+function emitStateChange(isCharging: boolean, level: number) {
+  const batteryState = getBatteryState(isCharging, level);
+  // prevent sending the same state change twice.
+  if (batteryState === lastReportedState) return;
   emitter.emit('Expo.batteryStateDidChange', { batteryState });
+  lastReportedState = batteryState;
```

Comment:
Might want to set this before we emit so that if there's a failure in a listener, we don't re-emit the same state.